nav2_util/include/nav2_util/geometry_utils.hpp

```diff
@@ -107,6 +124,25 @@ inline Iter min_by(Iter begin, Iter end, Getter getCompareVal)
   return lowest_it;
 }
 
+/**
+ * Find first element in iterator that is greater integrated distance than comparevalue
+ */
+template<typename Iter, typename Getter>
+inline Iter first_element_beyond(Iter begin, Iter end, Getter getCompareVal)
+{
+  if (begin == end) {
+    return end;
+  }
+  Getter dist = 0.0;
+  for (Iter it = begin; it != end - 1; it++){
```

Comment:
There are some linting problems here, check CircleCI to find them. You can reproduce locally if you run `ament_cpplint` and `ament_uncrustify` to have the right ROS2 formatting style